File: Ej42/Java/Figuras.java
import java.util.Scanner;
import java.lang.Math;

/*
 *Autor:  UnoCuatro
 */
class Figuras {
	private static double areaCirculo() {
		double r;
		Scanner sc=new Scanner(System.in);
		System.out.println("\tCirculo \n"
				+ "Radio: ");
		r=sc.nextDouble();
		
		return Math.PI * Math.pow(r, 2);
	}

	private static double areaCuadrado() {
		double l;

		Scanner sc=new Scanner(System.in);
		
		System.out.print("\tCuadrado\n"
				+ "Lado: ");
		l=sc.nextDouble();
		
		return Math.pow(l, 2);
	}

	private static double areaRectangulo() {
		double b,h;

		Scanner sc=new Scanner(System.in);
		
		System.out.print("\tRectangulo \n"
				+ "Base: ");
		b=sc.nextDouble();
		System.out.print("Altura: ");
		h=sc.nextDouble();
		
		return b * h;
	}

	private static double areaTriangulo() {
		double b,h;

		Scanner sc=new Scanner(System.in);
		
		System.out.print("\tTriangulo \n"
				+ "Base: ");
		b=sc.nextDouble();
		System.out.print("Altura: ");
		h=sc.nextDouble();

		return (b * h) / 2;
	}

	private static void menu(){
		short op;
		double area=0;
		boolean repetir=true;

		Scanner sc=new Scanner(System.in);
		
		do{
			System.out.println("\tCALCULAR AREA DE:\n"
					+ "1.-Circulo\n"
					+ "2.-Cuadrado\n"
					+ "3.-Rectangolo\n"
					+ "4.-Triangulo\n"
					+ "0.-Salir");
			op = sc.nextShort();

			System.out.println("");
			
			switch(op){
				case 1:
					area = areaCirculo();
					break;
				case 2:
					area = areaCuadrado();
					break;
				case 3:
					area = areaRectangulo();
					break;
				case 4:
					area = areaTriangulo();
					break;
			}

			if(op!=0){
				System.out.println("Area: " + area + "\n");
			} else {
				System.out.println("Fin");
				repetir = false;
			};

		}while(repetir);
	}

	public static void main (String[] args){
		menu();
	}
}

File: Ej42/C++/figuras.cpp
#include <iostream>
#include <cmath>

using namespace std;

/*
 * Autor: UnoCuatro
 */
float areaCirculo() {
	float r;
	cout << "\tCirculo" << endl
			<< "Radio: ";
	cin >> r;
	
	return M_PI * pow(r, 2);
}

float areaCuadrado() {
	float l;
	cout << "\tCuadrado" << endl
			<< "Lado: ";
	cin >> l;
	
	return pow(l, 2);
}

float areaRectangulo() {
	float b,h;
	cout << "\tRectangulo" << endl
			<< "Base: ";
	cin >> b;
	cout << "Altura: ";
	cin >> h;
	
	return b * h;
}

float areaTriangulo() {
	float b,h;
	cout << "\tTriangulo" << endl
			<< "Base: ";
	cin >> b;
	cout << "Altura: ";
	cin >> h;

	return (b * h) / 2;
}
void menu(){
	short op;
	float area;
	bool repetir=true;

	do{
		cout << "\tCALCULAR AREA DE: " << endl
				<< "1.-Circulo " << endl
				<< "2.-Cuadrado " << endl
				<< "3.-Rectangolo " << endl
				<< "4.-Triangulo" << endl
				<< "0.-Salir" << endl;
		cin >> op;
		cout << endl;

		switch(op){
			case 1:
				area = areaCirculo();
				break;
			case 2:
				area = areaCuadrado();
				break;
			case 3:
				area = areaRectangulo();
				break;
			case 4:
				area = areaTriangulo();
				break;
		}

		if(op!=0){
			cout << "Area: " << area << endl << endl;
		} else {
			cout << "Fin";
			repetir = false;
		};

	}while(repetir);
}

int main (){
	menu();
	return 0;
}